### DISCUSSION TITLE
MUI Autocomplete first stroke is not working

### DISCUSSION THREAD

**Discussion by dev-johnray** [react-hook-form/react-hook-form#2973] [Q&A] [answered]
There is a problem in Material-UI Autocomplete/Textfield. You need to type enter any character first before the error message disappears.

codesandbox: https://codesandbox.io/s/material-ui-autocomplete-with-longer-label-forked-1x3cj?from-embed=&file=/index.js

![error3](https://user-images.githubusercontent.com/49766928/93867564-cf1d0d80-fcfb-11ea-8222-551d4182e325.gif)

**Comment by bluebill1049** [comment, marked as answer]
please read the doc, you are mixing controller with uncontrolled... the moment you have value and onChange that's controlled.

I recommend you to read the Controller section

**Reply by dev-johnray** [reply]
can you correct my code? sorry, I'm only new in react. thank you.

**Reply by bluebill1049** [reply]
1. Take a look at this codesandbox: https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r
2. have a read at the doc: https://react-hook-form.com/api#Controller

**Reply by dev-johnray** [reply]
Got it! Thank you sir!

**Reply by bluebill1049** [reply]
learning something take time, if I just correct it, it's not the best way of learning things.
